### PR TITLE
[BEAM-367] Modify offset range tracker to use first response as start offset

### PR DESCRIPTION
**Pull request by ianzhou1**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [ ] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [ ] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [ ] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

### CODE REVIEW

File: sdks/java/core/src/test/java/org/apache/beam/sdk/io/range/OffsetRangeTrackerTest.java [hunk 1]
```diff
@@ -36,6 +37,16 @@
   @Rule public final ExpectedException expected = ExpectedException.none();
 
   @Test
+  public void testUpdateStartOffset() throws Exception {
+    OffsetRangeTracker tracker = new OffsetRangeTracker(100, 200);
```
Comment (dhalperi) [inline sdks/java/core/src/test/java/org/apache/beam/sdk/io/range/OffsetRangeTrackerTest.java]: assert that it is `100` here?

File: sdks/java/core/src/test/java/org/apache/beam/sdk/io/range/OffsetRangeTrackerTest.java [hunk 2]
```diff
@@ -145,14 +156,32 @@ public void testGetFractionConsumedDense() throws Exception {
   @Test
   public void testGetFractionConsumedSparse() throws Exception {
     OffsetRangeTracker tracker = new OffsetRangeTracker(100, 200);
-    assertEquals(0, tracker.getFractionConsumed(), 1e-6);
+    assertEquals(0.0, tracker.getFractionConsumed(), 1e-6);
+    assertTrue(tracker.tryReturnRecordAt(true, 100));
     assertTrue(tracker.tryReturnRecordAt(true, 110));
```
Comment (dhalperi) [inline sdks/java/core/src/test/java/org/apache/beam/sdk/io/range/OffsetRangeTrackerTest.java]: 0.0 before L161

File: sdks/java/core/src/test/java/org/apache/beam/sdk/io/range/OffsetRangeTrackerTest.java [hunk 3]
```diff
@@ -145,14 +156,32 @@ public void testGetFractionConsumedDense() throws Exception {
...
-    // Consumed positions through 110 = total 11 positions of 100.
-    assertEquals(0.11, tracker.getFractionConsumed(), 1e-6);
+    // Consumed positions through 110 = total 10 positions of 100.
```
Comment (dhalperi) [inline sdks/java/core/src/test/java/org/apache/beam/sdk/io/range/OffsetRangeTrackerTest.java]: `Consumed positions 100-109, total 10 positions of 100 in range.`

File: sdks/java/core/src/test/java/org/apache/beam/sdk/io/range/OffsetRangeTrackerTest.java [hunk 4]
```diff
@@ -128,13 +139,13 @@ public void testGetPositionForFractionDense() throws Exception {
   @Test
```
Comment (dhalperi) [inline sdks/java/core/src/test/java/org/apache/beam/sdk/io/range/OffsetRangeTrackerTest.java]: I'm surprised that the `getPositionForFraction` tests above didn't have to change after you removed the `+1`. Now that we're moved fractions slightly, we probably should also move the positions we infer from fractions, right?

Take a look and see if that is right.

**Comment by dhalperi** [comment]
R: @dhalperi

**Comment by ianzhou1** [comment]
Still have to fix getPositionForFractionConsumed()

**Pull request closed without merging**